website/blog/2025-02-12-best-alternative-to-mongodb-data-api-and-http-endpoints.md

```diff
@@ -0,0 +1,200 @@
+    ports:
+      - 27017:27017
+      - 8080:8080
+    environment:
+      - FERRETDB_POSTGRESQL_URL=postgres://username:password@postgres:5432/postgres
+      - FERRETDB_LISTEN_DATA_API_ADDR=:8080
+networks:
+  default:
+    name: ferretdb
+```
+
+Note that the FerretDB v2 release requires a PostgreSQL with DocumentDB extension as the backend.
+Once your FerretDB instance is running, the Data API endpoint will be available at `http://localhost:8080`.
+
+### Using FerretDB's Data API for CRUD operations
+
+Since you've enabled FerretDB's Data API (`FERRETDB_LISTEN_DATA_API_ADDR=:8080`), let's find, insert, update, and delete a single document using `curl`.
+
+#### Insert a document
+
+Insert a document into a collection called `users` in a `test` database.
+
+```sh
+curl -X POST http://127.0.0.1:8080/action/insertOne \
+  -H "Content-Type: application/json" \
+  -u username:password \
+  -d '{
+        "database": "test",
+        "collection": "users",
+        "document": {
+          "name": "Andrew",
+          "email": "andrew@example.com",
+          "age": 25
+        }
+      }'
+```
+
+Response:
+
+```json
+{ "n": 1.0 }
+```
+
+#### Find a document
+
+Next, let's retrieve the document we just inserted.
+
+```sh
+curl -X POST http://127.0.0.1:8080/action/find \
+  -H "Content-Type: application/json" \
+  -u username:password \
+  -d '{
+        "database": "test",
+        "collection": "users",
+        "filter": { "name": "Andrew" }
+      }'
+```
+
+Response:
+
+```json
+{
+  "documents": [
+    {
+      "_id": { "$oid": "67a2e9b70a5e9467a00918b0" },
+      "name": "Andrew",
+      "email": "andrew@example.com",
+      "age": 25
+    }
+  ]
+}
+```
+
+#### Update a document
+
+Let's update Andrew's email:
+
+```sh
+curl -X POST http://127.0.0.1:8080/action/updateOne \
+  -H "Content-Type: application/json" \
+  -u username:password \
+  -d '{
+        "database": "test",
+        "collection": "users",
+        "filter": { "name": "Andrew" },
+        "update": { "$set": { "email": "andrew.new@example.com" } }
+      }'
+```
+
+Response:
+
+```json
+{ "matchedCount": 1, "modifiedCount": 1 }
+```
+
+#### Delete a document
+
+And to complete the CRUD operations, let's delete Andrew's document:
+
+```sh
+curl -X POST http://127.0.0.1:8080/action/deleteOne \
+  -H "Content-Type: application/json" \
+  -u username:password \
+  -d '{
+        "database": "test",
+        "collection": "users",
+        "filter": { "name": "Andrew" }
+      }'
+```
+
+Response:
+
+```json
+{ "deletedCount": 1 }
+```
+
+## Easily replace MongoDB Data API with FerretDB
+
+With just a few months left before MongoDB's Data API and HTTPS endpoints reach end-of-life, it's crucial to migrate to a suitable alternative.
+FerretDB's Data API provides a seamless replacement, allowing developers to interact with their databases via HTTP.
+You can easily swap out your MongoDB instance for FerretDB and continue working with your data without any interruptions.
```

Comment:
I'd suggest to add a sentence in this section, or anywhere else where it suits best, that'd encourage users to contact us if they have issues with migration, or unsupported api features required by their use case.

Reply:
PTAL